app/gui/src/dashboard/components/dashboard/column/PathColumn.tsx

```diff
@@ -0,0 +1,235 @@
+/** @file A column displaying the path of the asset. */
+import FolderIcon from '#/assets/folder.svg'
+import FolderArrowIcon from '#/assets/folder_arrow.svg'
+import { Button, Popover, Text } from '#/components/AriaComponents'
+import SvgMask from '#/components/SvgMask'
+import { useEventCallback } from '#/hooks/eventCallbackHooks'
+import { useCategoriesAPI, useCloudCategoryList } from '#/layouts/Drive/Categories/categoriesHooks'
+import type { AnyCloudCategory } from '#/layouts/Drive/Categories/Category'
+import { useUser } from '#/providers/AuthProvider'
+import { useSetExpandedDirectoryIds, useSetSelectedKeys } from '#/providers/DriveProvider'
+import type { DirectoryId } from '#/services/Backend'
+import { isDirectoryId } from '#/services/Backend'
+import { Fragment, useTransition } from 'react'
+import invariant from 'tiny-invariant'
+import type { AssetColumnProps } from '../column'
+
+/** A column displaying the path of the asset. */
+export default function PathColumn(props: AssetColumnProps) {
+  const { item, state } = props
+
+  const { virtualParentsPath, parentsPath } = item
+
+  const { getAssetNodeById } = state
+
+  const { setCategory } = useCategoriesAPI()
+  const setSelectedKeys = useSetSelectedKeys()
+  const setExpandedDirectoryIds = useSetExpandedDirectoryIds()
+
+  // Path navigation exist only for cloud categories.
+  const { getCategoryByDirectoryId } = useCloudCategoryList()
+
+  // Parents path is a string of directory ids separated by slashes.
+  const splittedPath = parentsPath.split('/').filter(isDirectoryId)
```

Comment:
nit: technically the paste tense of `split` is `split`. this might be clearer though